packages/core/core-flows/src/order/workflows/cancel-order.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  FulfillmentDTO,
+  OrderDTO,
+  OrderWorkflow,
+  PaymentCollectionDTO,
+} from "@medusajs/types"
+import { MedusaError, deepFlatMap } from "@medusajs/utils"
+import {
+  WorkflowData,
+  createStep,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../common"
+import { cancelPaymentStep } from "../../payment/steps"
+import { deleteReservationsByLineItemsStep } from "../../reservation/steps"
+import { cancelOrdersStep } from "../steps/cancel-orders"
+import { throwIfOrderIsCancelled } from "../utils/order-validation"
+
+const validateOrder = createStep(
+  "validate-order",
+  ({
+    order,
+  }: {
+    order: OrderDTO
+    input: OrderWorkflow.CancelOrderWorkflowInput
+  }) => {
+    const order_ = order as OrderDTO & {
+      payment_collections: PaymentCollectionDTO[]
+      fulfillments: FulfillmentDTO[]
+    }
+
+    throwIfOrderIsCancelled({ order })
+
+    let refunds = 0
+    let captures = 0
+
+    deepFlatMap(order_, "payment_collections.payments", ({ payments }) => {
+      refunds += payments?.refunds?.length ?? 0
+      captures += payments?.captures?.length ?? 0
```

Comment:
thought: Guess this should be migrated to BigNumber no?

Reply:
This is only counting if there is any refund or capture. Not their amount.

Reply:
Ah sorry, I missed that 😅